Zend/zend_language_scanner.l

```diff
@@ -1971,6 +1971,33 @@ NEWLINE ("\r"|"\n"|"\r\n")
 	}
 }
 
+/* Bigint support */
+<ST_IN_SCRIPTING>{LNUM}"n" {
+	/* Length without the "n" suffix */
+	size_t len = yyleng - 1;
+	char *lnum = yytext;
+	zend_bool is_octal = lnum[0] == '0';
+
+	/* Digits 8 and 9 are illegal in octal literals. */
+	if (is_octal) {
```

Comment:
we should throw instead

Reply:
This does throw (this is C, not C++), and is based on the adjacent ordinary LNUM parsing. `zend_throw_exception(zend_ce_parse_error, "Invalid numeric literal", 0);`

For token_get_all, the choice to not throw without the TOKEN_PARSE flag is deliberate - tools such as php-parser couldn't work if token_get_all threw instead of returning the T_ERROR token.

Reply:
I mean to not allow octals in bigints at all - reject any input starting with `0` except `0n`

Reply:
And the token_get_all() implementation clears this exception - I was wondering how it worked for ordinary numbers

```
		/* Normal token_get_all() should not throw. */
		zend_clear_exception();
```

Reply:
> 
> 
> I mean to not allow octals in bigints at all - reject any input starting with `0` except `0n`

That makes  no sense, the GMP extension accepts Hex, Octal and Binary numbers just fine

Reply:
But usecases vs. possible mistakes? :)

Reply:
I'd find it more consistent to allow any integer token to have a bigint equivalent by adding the suffix `n` to it.

Possibly use cases would be cryptography with known hex values, arbitrary-precision math based on octal/hex/binary reference material, etc - e.g. https://en.wikipedia.org/wiki/Mersenne_prime